.github/workflows/pull-request.yml

```diff
@@ -60,9 +70,17 @@ jobs:
       with:
         node-version: 20
         cache: 'npm'
+    - name: Get changed files
+      id: changed-ui_tests
+      uses: tj-actions/changed-files@v45
+      with:
+        files: |
+          packages/**
+          tests/**
+          app/**
     - name: Install dependencies
+      if: steps.changed-ui_tests.outputs.any_changed == 'true'
       run: npm ci
-    - name: Run tests
-      run: npm test
```

Comment:
@frankieroberto Remove (mostly) duplicate test run here, though potentially given concurrent tests, you’d want this workflow to fail early if there’s a failing test before running expensive visual regression tests?